src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+
+    private int capacity;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        this.threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+        this.table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size == threshold) {
+            resize();
+        }
+
+        int keyHash = key == null ? 0 : key.hashCode();
+        int indexToPut = Math.abs(keyHash % capacity);
```

Comment:
extract private method
```suggestion
        int indexToPut = getIndex(key);
```